packages/typespec-client-generator-core/src/validate.ts

```diff
@@ -1,12 +1,42 @@
 import { Program } from "@typespec/compiler";
 import { createTCGCContext } from "./internal-utils.js";
 import { getAllModelsWithDiagnostics } from "./types.js";
+import { SdkModelType, SdkEnumType } from "./interfaces.js";
+import { reportDiagnostic } from "./lib.js";
 
 export function $onValidate(program: Program) {
   // Pass along any diagnostics that might be returned from the HTTP library
   const tcgcContext = createTCGCContext(program);
-  const [_, diagnostics] = getAllModelsWithDiagnostics(tcgcContext);
+  const [models, diagnostics] = getAllModelsWithDiagnostics(tcgcContext);
   if (diagnostics.length > 0) {
     program.reportDiagnostics(diagnostics);
   }
+
+  validateModels(program, models);
+}
+
+function validateModels(program: Program, models: (SdkModelType | SdkEnumType)[]) {
+  const modelNameMap = new Map<string, Set<SdkModelType | SdkEnumType>>();
```

Comment:
Btw we have DuplicateTracker until in the compiler that is quite usufruit for this

Reply:
applied